server/services/recategorisation/filter/recategorisationFilter.ts

```diff
@@ -244,9 +236,31 @@ export const filterListOfPrisoners = async (
           }
           break
         default:
-          throw new Error(`Invalid filter type: ${allFilters[i]}`)
+          throw new Error(`Invalid filter type: ${allFiltersWhichDoNotRequireFurtherDataToBeLoaded[i]}`)
       }
     }
     return true
   })
+
+  let offenderNumbersWithAdjudications = []
+  if (allFilters.includes(NO_ADJUDICATIONS_IN_THE_LAST_3_MONTHS)) {
+    const adjudicationsData = await loadAdjudicationsData(filteredPrisoners, nomisClient, agencyId)
+    offenderNumbersWithAdjudications = adjudicationsData.map(adjudicationsDatum => adjudicationsDatum.offenderNo)
+    filteredPrisoners = filteredPrisoners.filter(prisoner => {
+      return !offenderNumbersWithAdjudications.includes(prisoner.offenderNo)
+    })
+  }
+  let offenderNumbersWithLowRoshScore = []
+  if (allFilters.includes(LOW_ROSH)) {
+    offenderNumbersWithLowRoshScore = await getOffenderNumbersWithLowRoshScore(
```

Comment:
is this or line 247 slow?

if so, it could be worth just adding the functions to an array

```
  if (allFilters.includes(NO_ADJUDICATIONS_IN_THE_LAST_3_MONTHS)) {
    const adjudicationsDataPromise = loadAdjudicationsData(filteredPrisoners, nomisClient, agencyId)
  }
  if (allFilters.includes(LOW_ROSH)) {
    offenderNumbersWithLowRoshScorePromise =getOffenderNumbersWithLowRoshScore(...
}
```

then, run all the promises at the same time with 

```
const [adjudicationsData, offenderNumbersWithLowRoshScore] = await Promise.all([adjudicationsDataPromise, offenderNumbersWithLowRoshScorePromise])
```

requires jiggery pokery with the code after

but then everything runs rather than blocks and then runs and then blocks and then runs etc

Reply:
Thanks, good suggestion! I think the adjudications data is normally quite quick to load but this is still a good call... I'm going to add it in as part of a different PR though so I can run some of this logging on preprod sooner